src/background/service-worker-init.ts

```diff
@@ -49,13 +50,12 @@ async function initialize(): Promise<void> {
     const browserAdapterFactory = new BrowserAdapterFactory(userAgentParser);
     const logger = createDefaultLogger();
     const promiseFactory = createDefaultPromiseFactory();
-    const FIRE_AND_FORGET_EVENT_DELAY_MS = 2 * 60 * 1000; // 2 minutes
-    const browserEventManager = new BrowserEventManager(
-        promiseFactory,
-        logger,
-        FIRE_AND_FORGET_EVENT_DELAY_MS,
+    const browserEventProvider = new BrowserEventProvider();
+    const browserEventManager = new BrowserEventManager(promiseFactory, logger, true);
+    const browserAdapter = browserAdapterFactory.makeFromUserAgent(
```

Comment:
It would be good to leave a comment here explaining why it is important that this happens before any usage of `await` inside the initializer